Execute DOGM display-loop less often

AnHardt opened the pull request:
While reading: https://code.google.com/p/u8glib/wiki/tpictureloop
I got the suspicion that the content of the display loop is executed more frequent than expected.
Commit 1 proves that.
https://youtu.be/tdu7J0i-c4g
Commit 2 improves that a bit.
https://youtu.be/AEnBzdu9_tk
More to come.

AnHardt committed:
Move the 'alive dot' in every displayloop

Result see: https://www.youtube.com/watch?v=tdu7J0i-c4g
Loop is executed several times per display update.

AnHardt committed:
Move the condition before the loop

Result see: https://youtu.be/AEnBzdu9_tk
Dot is moving not so fast -> less executed loops.

AnHardt committed:
Testloop cleanup

Shifted condition should still be in.

AnHardt committed:
Revert "Testloop cleanup"

This reverts commit 6361404b91dcf8ba657b1f1e16d9717a8e280ae6.

AnHardt commented:
Sorry nonsens.

alexborro commented:
Humm, nice..

I think the optimal idea would be calling the lcd refresh function from an
interruption executed at about 1hz. There is no reason for a faster update.
And the buttons/scroller function must be called faster to make the display
responsive...

That way the LCD will be refreshed faster when navigating through menus,
but at a lower rate when at info screen... That will be awesome for Delta
bots.

2015-04-04 13:05 GMT-03:00 AnHardt notifications@github.com:

> While reading: https://code.google.com/p/u8glib/wiki/tpictureloop
> I got the suspicion that the content of the display loop is executed more
> frequent than expected.
> Commit 1 proves that.
> https://youtu.be/tdu7J0i-c4g
> Commit 2 improves that a bit.
> https://youtu.be/AEnBzdu9_tk
> 
> ## More to come.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/MarlinFirmware/Marlin/pull/1817
> Commit Summary
> - Move the 'alive dot' in every displayloop
> - Move the condition before the loop
> - Testloop cleanup
> 
> File Changes
> - _M_ Marlin/ultralcd.cpp
>   https://github.com/MarlinFirmware/Marlin/pull/1817/files#diff-0 (20)
> 
> Patch Links:
> - https://github.com/MarlinFirmware/Marlin/pull/1817.patch
> - https://github.com/MarlinFirmware/Marlin/pull/1817.diff
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/MarlinFirmware/Marlin/pull/1817.

## 

_"Não é o mais forte da espécie que sobrevive, nem o mais inteligente. É
aquele que se adapta melhor as mudanças"  ( Charles Darwin )_

Alex Borro

AnHardt committed:
Reduce display updates for dogm displays

In 'if (encoderPastThreshold || LCD_CLICKED)' lcdDrawUpdate was not set.

'lcd_status_update_delay'-loop was not effective in  function
lcd_status_screen().
The loop prevented the update of the status screen but not the displays
update.
Shifted the loop into lcd_update().

AnHardt commented:
In the previous version i only shifted the content of the loop out of the loop but did not count it anymore.

This time the update indeed will be only about once a second.

AnHardt commented:
And now travis can't find u8glib :-(

thinkyhead commented:
@AnHardt Feeling like this is ready to be merged?

AnHardt commented:
Yes.
